bb_server/src/particle/mod.rs

```diff
@@ -31,6 +31,24 @@ pub struct Particle {
 }
 
 impl Particle {
+  pub fn new(
+    ty: Type,
+    pos: FPos,
+    long_distance: bool,
+    offset: FPos,
+    count: u32,
+    data: f32,
+  ) -> Self {
+    Self {
+      ty,
+      pos,
+      long_distance,
+      offset,
+      count,
+      data,
+    }
+  }
+
```

Comment:
all the fields are public, no need for a constructor. Forcing users to use struct literals is also clearer, as you can see the field names.

Reply:
the constructor is still here. forgot to push maybe?